src/fabric_quality_factor.erl

```diff
@@ -0,0 +1,213 @@
+    true ->
+        {ok, {NewCounters, Acc}};
+    false ->
+        {error, Reason}
+    end;
+
+%% handle_message/3
+%%
+%% We need this callback/response coordinator to listen for a response from
+%% all healthy shards -- return the stop tuple only when all are accounted for.
+%% Inputs:
+%%   - {ok, ShardQFactor}: ShardQFactor is the integer per-shard quality
+%%     factor as determined for shard Shard.
+%%   - Shard: The open (shard) database. #shard.ref is used as the key for
+%%     the workers dict.
+%%   - {Counters, Acc}: Counters is the fabric_dict tracking per-shard
+%%     response state. Acc is our total quality factor so far.
+handle_message({ok, ShardQFactor}, Shard, {Counters, Acc}) ->
+    %% Store the per-shard quality factor in the workers fabric_dict
+    %% although we won't strictly need it there for computation.
+    C1 = fabric_dict:store(Shard, ShardQFactor, Counters),
+    case fabric_dict:any(nil, C1) of
+        true ->
+            {ok, {C1, Acc+ShardQFactor}};
+        false ->
+            {stop, Acc+ShardQFactor}
+    end;
+handle_message(_, _, Acc) ->
+    {ok, Acc}.
+
+
+%%%%%%%%%%%%%%%%%%%%%%
+%% fabric_rpc callback
+%%%%%%%%%%%%%%%%%%%%%%
+
+
+%% Calculate the shard sync quality factor
+%% The quality factor is an upper bound on the number of missing updates
+%% between shard copies for all copies of a shard.
+%% See: https://cloudant.fogbugz.com/f/cases/25615/Create-quality-factor-measurement-for-sequence#BugEvent.203899
+%% ShardName  - e.g. <<"shards/c0000000-ffffffff/ksnavely/motohacker.1368999314">>
+%% 
+%% Called by fabric_rpc:shard_sync_quality_factor
+shard_sync_quality_factor(ShardName) ->
+    DbName = mem3:dbname(ShardName),
+    Nodes = nodes_for_shard(ShardName, DbName),
+
+    %% For each TargetNode with a ShardName copy, find the upper bound of
+    %% updates to other shard replicas which are missing from TargetNode
+    MissingUpdates = [shard_sync_quality_factor(ShardName, TargetNode) ||
+        TargetNode <- Nodes],
+
+    %% The quality factor is the sum of possible missing updates
+    {ok, lists:sum(MissingUpdates)}.
+
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%% Internal functions for QF determination
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+
+%% Calculate the shard sync quality factor for a single node/shard copy. This
+%% is the number of updates to other shard copies not present on TargetNode
+%% ShardName  - e.g. <<"shards/c0000000-ffffffff/ksnavely/motohacker.1368999314">>
+%% TargetNode - e.g. 'dbcore@db11.julep.cloudant.net'
+shard_sync_quality_factor(not_found, _TargetNode) ->
+    not_found;
+shard_sync_quality_factor(ShardName, TargetNode) ->
+    DbName = mem3:dbname(ShardName),
+    Nodes = nodes_for_shard(ShardName, DbName),
+
+    %% Find the upper bound of updates to other shard copies which are missing
+    %% from TargetNode
+    MissingUpdates = [shard_missing_updates(Source, TargetNode, ShardName) ||
+        Source <- Nodes, Source =/= TargetNode],
```

Comment:
OK, here you're correctly filtering out `node()` as I described in my earlier deleted comment. This function is really the one you want to be calling as the RPC, using `node()` instead of any possible `TargetNode`.